test/unit/ngsiv2/examples/contextAvailabilityRequests/registerIoTAgent2.json

```diff
@@ -0,0 +1,20 @@
+{
+    "dataProvided": {
+        "entities": [
+            {
+                "type": "Motion",
+                "id": "Motion:motion1"
+            }
+        ],
+        "attrs": [
+            "moving"
+        ]
+    },
+    "provider": {
+        "http": {
+            "url": "http://smartGondor.com"
+        },
+        "legacyForwarding": true
```

Comment:
"legacyForwarding: true" means to use the same format that for NGSIv1. If this is expected to model a register for NGSIv2-based forwarding, then legacyForwarding should be omitted.

Reply:
In other words... I undesrtand that currently ITOAs are using "legacyForwarding: true" no matter NGSI version in config.js / env var. It should be the following way:

* If configuration specifies v1, then use "legacyForwarding: true" in registraitons
* If configuration specifies v2, then ommit in registrations

This behaviour should be properly covered by unit tests.

Reply:
> If configuration specifies v1, then use "legacyForwarding: true" in registraitons

I understand that in this case, the correct functionality is to keep on using the NGSIv1 so actually registrations rely on /NGSI9/registerContext and https://fiware-orion.readthedocs.io/en/1.6.0/user/walkthrough_apiv1/#ngsi9-standard-operations, right?

In the case of NGSIv2, I have done the required modifications with b4af787eec75b7e8657bbabb600f5ff3bb48fa00. The strange thing is that maybe when I implemented this, I did not understand correctly `legacyForwarding` and in the practice I implemented the new format to receive the notifications as you can see in https://github.com/telefonicaid/iotagent-node-lib/blob/master/lib/templates/notificationTemplateNgsi2.json and also in the examples used in the unit tests:
- https://github.com/telefonicaid/iotagent-node-lib/blob/master/test/unit/ngsiv2/examples/subscriptionRequests/bidirectionalNotification.json
- https://github.com/telefonicaid/iotagent-node-lib/blob/master/test/unit/ngsiv2/examples/subscriptionRequests/simpleNotification.json

Reply:
> I understand that in this case, the correct functionality is to keep on using the NGSIv1 so actually registrations rely on /NGSI9/registerContext and https://fiware-orion.readthedocs.io/en/1.6.0/user/walkthrough_apiv1/#ngsi9-standard-operations, right?

You are rigth.

> In the case of NGSIv2, I have done the required modifications with b4af787. The strange thing is that maybe when I implemented this, I did not understand correctly legacyForwarding and in the practice I implemented the new format to receive the notifications as you can see in https://github.com/telefonicaid/iotagent-node-lib/blob/master/lib/templates/notificationTemplateNgsi2.json and also in the examples used in the unit tests:

Not sure of understanding this... Anyway, just to clarify, notifications triggered by subscriptions and forwarded updates triggered by registrations use a different payload format. The references you cite (the template and the two examples) correspond to notifications. Fowarded updates will use the format described under `POST /v2/op/update` in the [NGSIv2 specification](https://fiware.github.io/specifications/ngsiv2/stable/).

Just to confirm it is working that way...

Reply:
Sorry, yesterday night I was too tired and I mixed registrations and notifications... It works as you explain. You can check the payload of forwarded updates in https://github.com/dcalvoalonso/iotagent-node-lib/blob/task/lazyAttsNgsiv2/test/unit/ngsiv2/lazyAndCommands/lazy-devices-test.js#172.

Reply:
Ok. NTC.